Jenkinsfile

```diff
@@ -2,99 +2,89 @@
 
 // Copyright (C) Pelagicore AB 2017
 
-// Helper function to run commands through vagrant
-def vagrant = {String command ->
-    sh "vagrant ssh -c \"${command}\""
-}
-
 /*
- * Supported values for bsp are "intel" or "rpi"
- * Supported values for qtauto are true or false
+ * For supported variant names, see conf/variants/ in meta-pelux
  */
-def buildManifest = {String variant_name, boolean bitbake_image ->
-    // Store the directory we are executed in as our workspace.
-    String yoctoDir = "/home/yoctouser/pelux_yocto"
-    String manifest = "pelux.xml"
+void buildManifest(String variant_name, String bitbake_image) {
 
-    // Everything we run here runs in a docker container handled by Vagrant
     node("DockerCI") {
 
-        // These could be empty, so check for that when using them.
-        environment {
-            YOCTO_CACHE_URL = "${env.YOCTO_CACHE_URL}"
-            YOCTO_CACHE_ARCHIVE_PATH = "${env.YOCTO_CACHE_ARCHIVE_PATH}"
-        }
-
-        // Stages are subtasks that will be shown as subsections of the finished
-        // build in Jenkins.
+        // We are going to need these later
+        String homeDir = "/home/yoctouser"
+        String yoctoDir = "${homeDir}/pelux_yocto"
+        String manifest = "pelux.xml"
 
         stage("Checkout ${variant_name}") {
-            // Checkout the git repository and refspec pointed to by jenkins
             checkout scm
-            // Update the submodules in the repository.
             sh 'git submodule update --init'
         }
 
-        stage("Start Vagrant ${variant_name}") {
-            // Start the machine (destroy it if present) and provision it
-            sh "vagrant destroy -f || true"
-            sh "vagrant up"
+        stage("Docker build ${variant_name}") {
+            // Build image from local Dockerfile with specified uid.
+            String uid = sh(returnStdout: true, script: "id -u").trim()
+            image = docker.build("pelux", "--build-arg userid=${uid} .")
```

Comment:
We need to set this so that the UID inside the docker container matches that of the Jenkins user, since the Yocto cache setup is currently depending on this to work (when rsyncing to the cache)

Reply:
Me and @rpannek spoke about changing this later so that it is done through the Jenkinsfile, but that would be a future change I'd say.

Reply:
Ok